### DISCUSSION TITLE
AppDynamics metric for AnalysisTemplate

### DISCUSSION THREAD

**Discussion by aamirmalek** [argoproj/argo-rollouts#3267] [General]
Hi 

I am trying to create a AnalysisTemplate based on the Analysis template but I am getting error. 

_Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(AnalysisTemplate.spec.metrics[0].provider): unknown field "appdynamics" in io.argoproj.v1alpha1.AnalysisTemplate.spec.metrics.provider_

is AppDynamics not a provider in the Argo-rollouts? If not, is there any other way to access appdynamics data?

**Comment by kostis-codefresh** [comment]
It is not a provider. 

But you should be able to use a webhook https://argo-rollouts.readthedocs.io/en/stable/analysis/web/ or job https://argo-rollouts.readthedocs.io/en/stable/analysis/job/ to get from AppDynamics what you want
